src/hotspot/share/gc/g1/g1Policy.cpp

```diff
@@ -890,6 +893,17 @@ void G1Policy::record_young_collection_end(bool concurrent_operation_is_full_mar
     }
     _analytics->report_card_scan_to_merge_ratio(scan_to_merge_ratio, is_young_only_pause);
 
+    // Update prediction for code root scan
+    size_t const total_code_roots_scanned = p->sum_thread_work_items(G1GCPhaseTimes::CodeRoots, G1GCPhaseTimes::CodeRootsScannedNMethods) +
+                                            p->sum_thread_work_items(G1GCPhaseTimes::OptCodeRoots, G1GCPhaseTimes::CodeRootsScannedNMethods);
+
+    if (total_code_roots_scanned > G1NumCodeRootsCostSampleThreshold) {
+      double avg_time_code_root_scan = average_time_ms(G1GCPhaseTimes::CodeRoots) +
+                                       average_time_ms(G1GCPhaseTimes::OptCodeRoots);
+
+      _analytics->report_cost_per_code_root_scan_ms(avg_time_code_root_scan / total_code_roots_scanned, is_young_only_pause);
```

Comment:
It's kind of a preexisting issue -- it's super weird to see `avg_time / total_count`; `sum_time / total_count` is more reasonable. The current calculation assumes the same number of gc-workers is used for each gc-pause, I believe.

Reply:
> sum_time / total_count is more reasonable

doesn't account for work done in parallel.

Reply:
OK, maybe `set_young_collection_default_active_worker_threads` and/or its surrounding logic could be revised. It suggests that the number of workers can vary for each young GC, while predictors assume that the number of workers is constant. Just a cursory remark on this preexisting inconsistency, as my initial understanding was that a predictor tracks time/work-unit per thread, rather than time/work-unit per N threads.

Yes, it's a separate issue.